internal/testrunner/runners/system/tester.go

```diff
@@ -937,7 +940,7 @@ func (r *tester) prepareScenario(ctx context.Context, config *testConfig, svcInf
 
 	// Input packages can set `data_stream.dataset` by convention to customize the dataset.
 	dataStreamDataset := ds.Inputs[0].Streams[0].DataStream.Dataset
-	if scenario.pkgManifest.Type == "input" {
+	if scenario.pkgManifest.Type == "input" || dataStreamOverrides[scenario.pkgManifest.Name] {
```

Comment:
I wonder if we should do this in all cases and not only for input packages :thinking: Though integration packages should not need to configure datasets.

Or maybe this `amazon_security_lake` package should be an input package if it is expected to be used with custom datasets?

Reply:
Adding this for all integration packages seems unnecessary and might unexpected results in tests. The thing is with this specific integration, there's a central data stream that runs the tests, but then re routes the results to a separate data stream. This re route mechanism causes the system test to fail. In future other integration packages might also do this, hence why I created this override list.

This also cannot be an input package cause it has multiple data streams and ingest pipelines had have a lot of mapping logic. Generally input packages just consume raw inputs in a single generic data stream and dump them into elastic search.

Reply:
> The thing is with this specific integration, there's a central data stream that runs the tests, but then reroutes the results to a separate data stream.

I see we have some packages with routing rules, but don't have system tests, it would be good to identify what is missing and fix it so all these packages can have system tests.

On the other hand I see that the `entityanalytics_entra_id` package has routing rules and system tests. Is it doing some trick to workaround the current limitations? Or is this package doing something we could apply in other cases?

> Maybe we can add support for a new package spec variable, whose presence could trigger this ? Then we can remove the list.

Yes, I would prefer this before adding a list with exceptions, but lets try to identify first what is missing, or what other packages are doing to have system tests with routing rules.

I think it would be good to open an issue identifying the problem, and then we could discuss about possible solutions. @ShourieG would this work for you?

Reply:
@jsoriano I saw the **entityanalytics_entra_id** integration, and I don't think the systems tests over there are working properly, if you see the sample_event.json, it only contains agent data and does not contain any fields under the **entityanalytics_entra_id** namespace as per the field mappings. It also seems that the entityanalytics input has some internal mechanism to publish logs to all datasets which is not supported by the aws input, hence why a sample_event.json is even being generated in the first place even though the contents are incorrect.